ambry-protocol/src/test/java/com.github.ambry.protocol/RequestResponseTest.java

```diff
@@ -133,17 +136,29 @@ public long writeTo(WritableByteChannel channel)
 }
 
 public class RequestResponseTest {
+  private Random random = new Random();
+
   private void testPutRequest(MockClusterMap clusterMap, int correlationId, String clientId, BlobId blobId,
       BlobProperties blobProperties, byte[] userMetadata, BlobType blobType, byte[] blob, int blobSize)
       throws IOException {
     int sizeInBlobProperties = (int) blobProperties.getBlobSize();
     PutRequest request = new PutRequest(correlationId, clientId, blobId, blobProperties, ByteBuffer.wrap(userMetadata),
         new ByteBufferInputStream(ByteBuffer.wrap(blob)), blobSize, blobType);
     ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
-    WritableByteChannel writableByteChannel = Channels.newChannel(outputStream);
+    // Create channels in such a way that writeTo() may not be able to write out all the data at once.
+    int allocationSize = random.nextInt((int) request.sizeInBytes()) + 1;
```

Comment:
Did think of that but just felt that it would be an overkill to add more tests. It is highly unlikely that we would miss the case where we cannot write out all data at once. Being able to send all data at once is different, but so are the cases where total bytes is not multiple of the channel capacity and such. Rather than test each such case explicitly, I felt that randomizing is effective enough.